src/ifcgeom/IfcGeomIteratorImplementation.h

```diff
@@ -590,6 +494,112 @@ namespace IfcGeom {
         const gp_XYZ& bounds_max() const { return bounds_max_; }
 
 	private:
+		void addRepresentationsFromContextIds() {
+			for (auto context_id : settings.context_ids()) {
+				IfcSchema::IfcGeometricRepresentationContext* context = ifc_file->instance_by_id(context_id)->as<IfcSchema::IfcGeometricRepresentationContext>();
+				representations->push(context->RepresentationsInContext());
+
+				try {
+					if (context->Precision() && *context->Precision() < lowest_precision_encountered) {
```

Comment:
`Precision` is a derived attribute in case of sub contexts. In the other 'branch' we have guaranteed we aren't iterating over subcontexts for precision, but here I think there is a good chance it is a subcontext (?). Unforunately we have to 'implement' the derived attribute manually so `if (context->as<>(IfcGeometricRepresentationSubContext)) use context->as<>(IfcGeometricRepresentationSubContext)->ParentContext()->Precision()`